### PR TITLE
Working on array navigation and size

### PR DESCRIPTION
**Pull request by pavlopoul**

### CODE REVIEW

File: vxquery-core/src/main/java/org/apache/vxquery/xmlquery/ast/ArgumentNode.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.apache.vxquery.xmlquery.ast;
+
+import java.util.List;
+
+import org.apache.vxquery.util.SourceLocation;
+
+public class ArgumentNode extends ASTNode {
```
Comment (prestoncarman) [inline vxquery-core/src/main/java/org/apache/vxquery/xmlquery/ast/ArgumentNode.java:23]: Is this the Argument List we talked about? Why call it ArgumentNode or ARGUMENT_EXPRESSION. Confused about the naming here.

File: vxquery-core/src/main/java/org/apache/vxquery/xmlquery/translator/XMLQueryTranslator.java [hunk 2]
```diff
@@ -2022,6 +2054,10 @@ private ILogicalExpression data(ILogicalExpression expr) throws SystemException
         return new ScalarFunctionCallExpression(BuiltinFunctions.FN_DATA_1, Collections.singletonList(mutable(expr)));
     }
 
+    private ILogicalExpression size(ILogicalExpression expr) throws SystemException {
```
Comment (prestoncarman) [inline vxquery-core/src/main/java/org/apache/vxquery/xmlquery/translator/XMLQueryTranslator.java:2057]: Is this used? I think this can be removed.

File: vxquery-xtest/src/test/resources/ExpectedTestResults/Json/Array/q03_array.txt [hunk 3]
```diff
@@ -1 +1 @@
-[ [ 1, 2 ], [ 2, 3 ] ]
\ No newline at end of file
+[[1,2],[2,3]]
```
Comment (prestoncarman) [inline vxquery-xtest/src/test/resources/ExpectedTestResults/Json/Array/q03_array.txt:1]: I thought this change was already in master!?!?!

File: vxquery-xtest/src/test/resources/cat/JsonArrayNavigationQueries.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      <output-file compare="Text">q01_array_navigation.txt</output-file>
+   </test-case>
+   <test-case name="json-array-navigation-q02" FilePath="Json/Array/Navigation" Creator="Christina Pavlopoulou">
+      <description>Navigate through an array.</description>
+      <query name="q02_array_navigation" date="2016-06-16"/>
+      <output-file compare="Text">q02_array_navigation.txt</output-file>
+   </test-case>
+   <test-case name="json-array-navigation-q03" FilePath="Json/Array/Navigation" Creator="Christina Pavlopoulou">
+      <description>Size of an array.</description>
+      <query name="q03_array_navigation" date="2016-06-16"/>
```
Comment (prestoncarman) [inline vxquery-xtest/src/test/resources/cat/JsonArrayNavigationQueries.xml:35]: Add a test case with an empty array.

File: vxquery-xtest/src/test/resources/cat/JsonArrayNavigationQueries.xml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+<test-group xmlns="http://www.w3.org/2005/02/query-test-XQTSCatalog" name="JsonArrayNavigationQueries" featureOwner="VXQuery">
+   <GroupInfo>
+      <title>Json Array Navigation</title>
+      <description/>
+   </GroupInfo>
+   <test-case name="json-array-navigation-q01" FilePath="Json/Array/Navigation" Creator="Christina Pavlopoulou">
+      <description>Navigate through an array.</description>
+      <query name="q01_array_navigation" date="2016-06-16"/>
+      <output-file compare="Text">q01_array_navigation.txt</output-file>
```
Comment (prestoncarman) [inline vxquery-xtest/src/test/resources/cat/JsonArrayNavigationQueries.xml:26]: What happens if you ask for an index outside of the array? Create a test case and throw the correct error.

File: vxquery-xtest/src/test/resources/ExpectedTestResults/Json/Array/q03_array.txt (hunk 3 above)
Comment (pavlopoul) [inline vxquery-xtest/src/test/resources/ExpectedTestResults/Json/Array/q03_array.txt:1]: The one that is in master is with whitespace...now I removed whitespaces

File: vxquery-core/src/main/javacc/xquery-grammar.jj [hunk 6]
```diff
@@ -1743,17 +1744,54 @@ NameTestNode Wildcard():
...
+    }
+}
+
+ASTNode Arguments()    :
+{
+    List<ASTNode> expr = new ArrayList<ASTNode>();
+    Token start;
+}
+{
+   start= "(" expr=ArgumentList(){
```
Comment (prestoncarman) [inline vxquery-core/src/main/javacc/xquery-grammar.jj:1772]: whitespace

File: vxquery-core/src/main/javacc/xquery-grammar.jj [hunk 7]
```diff
@@ -1743,17 +1744,54 @@ NameTestNode Wildcard():
...
-    expr = PrimaryExpr() pList = PredicateList() {
-        FilterExprNode fe = new FilterExprNode(expr.getSourceLocation());
-        fe.setExpr(expr);
-        fe.setPredicates(pList);
-        return fe;
+    expr = PrimaryExpr() (arguments = Predicate() {
+        argList.add(arguments);
+        } | arguments = Arguments() {
+            argList.add(arguments);
+        })*{
```
Comment (prestoncarman) [inline vxquery-core/src/main/javacc/xquery-grammar.jj:1758]: Indentation seems off. Make sure opening and closing braces line up on each line.
